### PR TITLE
[SYNPY-1482] Updating to remove loop executor during annotation store

### PR DESCRIPTION
**Pull request by BryanFauble**
**Problem:**

1. Storage of annotation was taking place in a loop executor

**Solution:**

1. Updating logic to be stored without the executor/other threads

**Testing:**

1. Unit/Integration tests

### CODE REVIEW

File: synapseclient/api/annotations.py [hunk 1]
```diff
@@ -47,3 +47,36 @@ def set_annotations(
...
+            the `.login()` method.
+
+    Returns: The annotations set in Synapse.
+    """
+    annotations_dict = asdict(annotations)
+
+    synapse_annotations = _convert_to_annotations_list(
+        annotations_dict["annotations"] or {}
+    )
+    from synapseclient import Synapse
```
Comment (linglp) [inline synapseclient/api/annotations.py:71]: why here the import statement is in the function call?

Comment (BryanFauble) [reply synapseclient/api/annotations.py:71]: This is to prevent circular import issues. This is a "lazy" import. It was the easiest way to fix the problem, but I'd be open to suggestions.

File: synapseclient/api/__init__.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 __all__ = [
     # annotations
     "set_annotations",
+    "set_annotations_async",
```
Comment (linglp) [inline synapseclient/api/__init__.py:33]: I am also curious what is the purpose of "__all__" here and why did you add set_annotations_async?

Comment (BryanFauble) [reply synapseclient/api/__init__.py:33]: This is defining what can be imported from `synapseclient.api`:

https://www.geeksforgeeks.org/python-__all__/

It allows you to do:
`from synapseclient.api import set_annotations_async` instead of `from synapseclient.api.annotations import set_annotations_async`

File: synapseclient/api/annotations.py (hunk 1 above)
Comment (linglp) [reply synapseclient/api/annotations.py:71]: Thanks. I actually have never heard of lazy import before. I looked it up and it seems like the benefit of lazy import is to load a module when the module is being used. But`Synapse` is so commonly used which makes me wonder how big the benefit is.. But thanks for teaching me lazy import, I learned something new today!

File: synapseclient/api/__init__.py (hunk 2 above)
Comment (linglp) [reply synapseclient/api/__init__.py:33]: wow thank you for the explanation Bryan!

File: synapseclient/api/annotations.py (hunk 1 above)
Comment (thomasyu888) [reply synapseclient/api/annotations.py:71]: @linglp to circle back around to this.  It's important to note that we use lazy importing to avoid circular imports.

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!
